django/contrib/gis/db/backends/base/operations.py

```diff
@@ -104,7 +104,12 @@ def get_geom_placeholder(self, f, value, compiler):
 
     def check_aggregate_support(self, aggregate):
         if aggregate.contains_aggregate == 'gis':
```

Comment:
Hmm right. I don't really agree with that change FWIW. There should be a better way of checking whether or not the aggregate is allowed.

Other than cleanliness (I find multi-valued properties a bit of a code smell), if geo aggregates learn to support Combinable `Extent()+Extent()` then the code is going to fail - because `contains_aggregate` is going to return `True` rather than `gis`.

If you look at `refs_field` in `aggregates.py`, there is already a helper method for comparing allowed expressions with input fields. SQLite makes use of this by providing a list of disallowed aggregate/field combinations, which also works with nested expressions.

I also think (and this would be a change that's probably backwards-incompatible) that the backends should be passing a list of disallowed aggregates rather than allowed. Then users are free to create their own GeoAggregates that will work, rather than requiring to be on a whitelist. I don't know if this would be a feature GIS users would want though (are there any other geo aggregate types?).